flowzone.yml

```diff
@@ -1271,78 +1278,96 @@ jobs:
 
       # https://github.com/orgs/community/discussions/50055
       # https://www.levibotelho.com/development/commit-a-file-with-the-github-api/
+      # https://octokit.github.io/rest.js/v21/#git-create-blob
+      # https://octokit.github.io/rest.js/v21/#git-create-tree
       - name: Create blobs and tree objects
         if: inputs.disable_versioning != true
         id: create_tree
-        shell: bash
         env:
-          GH_TOKEN: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
           PARENT_COMMIT_SHA: ${{ steps.git_describe.outputs.sha }}
-        run: |
-          # Temporary array to hold our new tree objects
-          declare -a tree_array
-
-          # Use git status to check for new and modified files
-          modified_files="$(git diff --name-only ; git ls-files --others --exclude-standard)"
-
-          # Extract changes
-          for file in $modified_files; do
-
-            echo "Creating blob of file $file..."
-            base64 -w0 "$file" > content.base64
-            response="$(gh api -X POST "/repos/$GH_REPO/git/blobs" \
-              -F 'content=@content.base64' \
-              -F 'encoding=base64')"
-            rm content.base64
-
-            echo "$response" | jq .
-            blob_sha="$(echo "$response" | jq -r .sha)"
-
-            # Add blob to our tree
-            tree_array+=("{\"path\":\"$file\",\"mode\":\"100644\",\"type\":\"blob\",\"sha\":\"$blob_sha\"}")
-          done
-
-          # Get the SHA of the tree the parent commit points to
-          base_tree_sha="$(git show -s --format=%T "$PARENT_COMMIT_SHA")"
-
-          # Create JSON array for tree creation
-          tree_json="$(printf ",%s" "${tree_array[@]}")"
-          tree_json="${tree_json:1}"
-          tree_json="[$tree_json]"
-          tree_json="{\"tree\": $tree_json, \"base_tree\": \"$base_tree_sha\"}"
-
-          echo "Creating tree..."
-          echo "$tree_json" | jq .
-
-          response="$(echo "$tree_json" | gh api -X POST "/repos/$GH_REPO/git/trees" --input -)"
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7
+        with:
+          github-token: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
+          script: |
+            const { execSync } = require('child_process');
+            const fs = require('fs');
+
+            // Get modified and untracked files
+            const getModifiedFiles = () => {
+              const modifiedCmd = 'git diff --name-only';
+              const untrackedCmd = 'git ls-files --others --exclude-standard';
+              const modified = execSync(modifiedCmd).toString().trim().split('\n');
+              const untracked = execSync(untrackedCmd).toString().trim().split('\n');
+              return [...new Set([...modified, ...untracked])].filter(f => f);
```

Comment:
What are you filtering here? Null lines? Blank lines?

Reply:
Both, either, I just have trust issues processing git command line output.

Is there a better way?

Reply:
No, it's fine to filter, I'm just asking because I don't totally know the commands output. This is filtering nullish values, including empty strings, so it is technically correct. 

If you also wanted to filter blank lines (all whiltespace), you would have to do something like `f != null && !!f.trim()`

Reply:
But the current still filters empty strings, undefined, and null, correct? Just not whitespace?

Reply:
Yes, that's correct